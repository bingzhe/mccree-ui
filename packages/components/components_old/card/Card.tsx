import * as React from "react";
import classNames from "classnames";
import { ConfigContext } from "../config-provider";
import CardContent from "./CardContent";
import CardFooter from "./CardFooter";
import { hasChild, pickChild } from "../utils/collection";
import { CardFC, defaultProps } from "./Card.type";

const { useContext } = React;

const Card: CardFC = (props) => {
    const {
        variant,
        type,
        shadow,
        width,
        style: styleProp,
        hoverable,
        className,
        children,
        ...restProps
    } = props;

    const [withoutFooterChildren, footerChildren] = pickChild(children, CardFooter);
    const hasContent = hasChild(withoutFooterChildren, CardContent);

    const { getPrefixCls } = useContext(ConfigContext);
    const prefixCls = getPrefixCls("card");

    const classes = classNames(prefixCls, className, {
        [`${prefixCls}-${variant}`]: variant,
        [`${prefixCls}-${type}`]: type,
        [`${prefixCls}-shadow`]: shadow,
        [`${prefixCls}-hoverable`]: hoverable
    });

    const style = {
        ...styleProp,
        width: width
    };

    return (
        <div className={classes} style={style} {...restProps}>
            {hasContent ? (
                withoutFooterChildren
            ) : (
                <CardContent>{withoutFooterChildren}</CardContent>
            )}
            {footerChildren}
        </div>
    );
};

Card.Content = CardContent;
Card.Footer = CardFooter;
Card.Actions = CardFooter;
Card.displayName = "MR_Card";
Card.defaultProps = defaultProps;

export default Card;
