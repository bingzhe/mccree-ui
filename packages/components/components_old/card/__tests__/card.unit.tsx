import * as React from "react";
import { render, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Card from "../index";

describe("Card", () => {
    it("should render correctly", () => {
        const wrapper = render(
            <>
                <Card>xx</Card>
                <Card variant="outlined">xx</Card>
                <Card shadow>xx</Card>
                <Card hoverable>xx</Card>
                <Card type="primary">xx</Card>
                <Card type="secondary">xx</Card>
                <Card type="success">xx</Card>
                <Card type="info">xx</Card>
                <Card type="warning">xx</Card>
                <Card type="error">xx</Card>
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("should render Card.Footer/Card.Actions/Card.Content correctly", () => {
        const wrapper = render(
            <>
                <Card>
                    <Card.Content>XX</Card.Content>
                    <Card.Footer>YY</Card.Footer>
                </Card>
                <Card>
                    <Card.Content>XX</Card.Content>
                    <Card.Actions>YY</Card.Actions>
                </Card>
            </>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it("shoulid find mccree-card classnames", () => {
        const wrapper = mount(<Card>xx</Card>);
        expect(wrapper.find(".mccree-card").length).toBe(1);
    });
    it("shoulid find mccree-card-hoverable classnames", () => {
        const wrapper = mount(<Card hoverable>xx</Card>);
        expect(wrapper.find(".mccree-card-hoverable").length).toBe(1);
    });
    it("shoulid find mccree-card-shadow classnames", () => {
        const wrapper = mount(<Card shadow>xx</Card>);
        expect(wrapper.find(".mccree-card-shadow").length).toBe(1);
    });
    it("shoulid find mccree-card-outlined classnames", () => {
        const wrapper = mount(<Card variant="outlined">xx</Card>);
        expect(wrapper.find(".mccree-card-outlined").length).toBe(1);
    });
});
