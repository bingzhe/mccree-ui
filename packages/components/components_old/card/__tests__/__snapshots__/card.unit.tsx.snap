// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card.Footer/Card.Actions/Card.Content correctly 1`] = `
Array [
  <div
    class="mccree-card mccree-card-contained"
  >
    <div
      class="mccree-card-content"
    >
      XX
    </div>
    <footer
      class="mccree-card-footer"
    >
      YY
    </footer>
  </div>,
  <div
    class="mccree-card mccree-card-contained"
  >
    <div
      class="mccree-card-content"
    >
      XX
    </div>
    <footer
      class="mccree-card-footer"
    >
      YY
    </footer>
  </div>,
]
`;

exports[`Card should render correctly 1`] = `
Array [
  <div
    class="mccree-card mccree-card-contained"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-outlined"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-shadow"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-hoverable"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-primary"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-secondary"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-success"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-info"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-warning"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
  <div
    class="mccree-card mccree-card-contained mccree-card-error"
  >
    <div
      class="mccree-card-content"
    >
      xx
    </div>
  </div>,
]
`;
